### PR TITLE
Make Converter classes also catch ArgumentNullException

### LINKED ISSUE #1863: Make Converter classes also catch ArgumentNullException
as I noticed that the null pointer catch doesn't actually trigger, instead the non existing key read from JObject seems to result in argument null exception instead.

### PR DESCRIPTION
**Pull request by Apsum**
closes #1863

### CODE REVIEW

File: src/saving/serializers/ThriveJsonConverter.cs [hunk 1]
```diff
@@ -686,6 +686,11 @@ private void RunPostPropertyDeserializeActions(object instance)
             throw new JsonException($"no suitable constructor found for current type: {objectType}");
         }
 
+        if ( item == null)
```
Comment (Kasterisk) [inline src/saving/serializers/ThriveJsonConverter.cs:689]: There's a whitespace in front of the first bracket.

**Review by Kasterisk — CHANGES_REQUESTED** [review]
I'm assuming this would work as intended. One code formatting issue:

File: src/saving/serializers/ThriveJsonConverter.cs [hunk 2]
```diff
@@ -686,6 +686,11 @@ private void RunPostPropertyDeserializeActions(object instance)
...
+        {
+            throw new ArgumentNullException($" Non existing key from JObject: {item} ");
```
Comment (hhyyrylainen) [inline src/saving/serializers/ThriveJsonConverter.cs:691]: Umm, the issue is about *catching* in the various converters like GodotColourConverter, GodotBasisConverter etc. and not about throwing here. And btw `item` is already used on line 670 so this check against null here is not very useful.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

**Comment by hhyyrylainen** [comment]
@FuJa0815 did you revert all changes done by @Apsum ? If so, I think you should make a new PR instead...

**Pull request closed without merging**